lib/ssl/doc/src/ssl.xml

```diff
@@ -343,11 +343,33 @@
     <datatype>
       <name name="key"/>
       <desc>
-	<p>The DER-encoded user's private key or a map referring to a crypto
-	engine and its key reference that optionally can be password protected,
-	see also <seemfa marker="crypto:crypto#engine_load/3"> crypto:engine_load/3
-	</seemfa> and <seeguide marker="crypto:engine_load"> Crypto's Users Guide</seeguide>. If this option 
-	is supplied, it overrides option <c>keyfile</c>.</p>
+	
+	<p>The user's private key. The map formats referring to a
+	crypto engine/provider or Erlang fun together with key reference
+	information, can both be used for customized signing with
```

Comment:
this `together with key reference information` part refers to engine but is placed after Erlang fun ... which is confusing... 

this would be easier to understand maybe:
_The user's private key. The map formats referring to a crypto engine/provider(with key reference information) or Erlang fun, can both be used for customized signing with for instance hardware security modules (HSM) or trusted platform modules (TPM)._

Reply:
@u3s I reworded it as you suggested. The reason for the original formulation was that I am thinking maybe even the fun version could want to provide own options. This is not implemented right now but could easily be added.